packages/api/src/external/carequality/command/cq-directory/create-or-update-cq-organization.ts

```diff
@@ -0,0 +1,51 @@
+import { CarequalityManagementAPI } from "@metriport/carequality-sdk";
+import { makeCarequalityManagementAPI } from "../../api";
+import { CQOrganization } from "../../organization";
+import { CQOrgDetails } from "../../shared";
+
+const cq = makeCarequalityManagementAPI();
+
+export async function createOrUpdateCQOrganization(orgDetails: CQOrgDetails): Promise<void> {
+  if (!cq) throw new Error("Carequality API not initialized");
+  const org = CQOrganization.fromDetails(orgDetails);
+  const cqOrgString = org.toXmlString();
+  const orgExists = await doesCQOrganizationExist(org.oid);
+  if (orgExists) {
+    await updateCQOrganization(cq, cqOrgString, org.oid);
+  } else {
+    await registerOrganization(cq, cqOrgString);
+  }
+}
+
+async function doesCQOrganizationExist(oid: string): Promise<boolean> {
+  if (!cq) throw new Error("Carequality API not initialized");
+  const org = await cq.listOrganizations({ count: 1, oid });
+  if (org.length > 0) {
+    return true;
+  }
+  return false;
+}
+
+async function updateCQOrganization(
+  cq: CarequalityManagementAPI,
+  cqOrg: string,
```

Comment:
Why does this and the one below receive a string instead of an instance of `CQOrganization`? Try to keep strong types around as much as possible.

Reply:
Sounds good